### PR TITLE
qemu.tests.balloon_service: Some updates for the code.

### PR DESCRIPTION
**Pull request by CongLi**
1. Sleep a while before checking memory.
2. Update the way for comparing memory.

Signed-off-by: Cong Li <coli@redhat.com>

id: 1231037

### CODE REVIEW

**Comment by CongLi** [comment]
01:43:55 INFO | PASS 1-smp_4.8192m.repeat1.run_test.Host_RHEL.m7.u3.raw.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_service

**Comment by huangyum** [comment]
Test failed on my host, already apply #763 .

2016-11-02 03:24:38,960 stacktrace       L0038 ERROR|
2016-11-02 03:24:38,960 stacktrace       L0041 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py:420
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR| Traceback (most recent call last):
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|     self._runTest()
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|     run_func(self, params, env)
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|     return fn(*args, **kwargs)
2016-11-02 03:24:38,985 stacktrace       L0044 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_service.py", line 161, in run
2016-11-02 03:24:38,986 stacktrace       L0044 ERROR|     memory_check(vm, get_polling_output, 'stat-total-memory')
2016-11-02 03:24:38,986 stacktrace       L0044 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/balloon_service.py", line 98, in memory_check
2016-11-02 03:24:38,986 stacktrace       L0044 ERROR|     check_mem_ratio))
2016-11-02 03:24:38,986 stacktrace       L0044 ERROR| TestFail: stat-total-memory of guest 4096 is not equal to 0 in qmp, the acceptable ratio is 0.1
2016-11-02 03:24:38,986 stacktrace       L0045 ERROR|
2016-11-02 03:24:38,987 test             L0589 ERROR| FAIL 4-smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_service -> TestFail: stat-total-memory of guest 4096 is not equal to 0 in qmp, the acceptable ratio is 0.1

**Comment by CongLi** [comment]
Thanks @huangyum .

For this case, you need install balloon driver at first.
You can run single_driver_install.with_balloon

**Comment by huangyum** [comment]
Also hit this:
03:22:54 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py:420
03:22:54 ERROR| Traceback (most recent call last):
03:22:54 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-42.0-py2.7.egg/avocado_vt/test.py", line 206, in runTest
03:22:54 ERROR|     raise exceptions.TestFail(details)
03:22:54 ERROR| TestFail: Balloon test failed after enlarge memory    [context: (run_ballooning_test) --> Check memory status after enlarge memory]
03:22:54 ERROR|
03:22:54 ERROR| FAIL 3-smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.io-github-autotest-qemu.balloon_service -> TestFail: Balloon test failed after enlarge memory    [context: (run_ballooning_test) --> Check memory status after enlarge memory]

**Comment by huangyum** [comment]
@CongLi  Thanks. 
After installing balloon driver, test passed with Win8.1 guest, but failed with win8 guest.   The error message is in above comment.

**Comment by huangyum** [comment]
Test failed with both win8 and win8.1 guest. For win8.1 guest, sometimes the test passed. 
(1/2) smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.io-github-autotest-qemu.balloon_service: FAIL (330.26 s)
 (2/2) smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.balloon_service: FAIL (375.59 s)

The error message is in above comment. 
More debug log:
04:48:29 DEBUG| (monitor qmp1) Sending command 'balloon'
04:48:29 DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 3835691008}, 'id': 'PRPeII03'}
04:48:30 INFO | BALLOON_CHANGE QMP event received
04:48:30 DEBUG| (monitor qmp1) Sending command 'query-balloon'
04:48:30 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'CkXGb9hY'}
04:48:30 DEBUG| (monitor qmp1) Response to 'query-balloon' (re-formated)
04:48:30 DEBUG| (monitor qmp1)    actual: 3835691008
04:48:30 INFO | Context: (run_ballooning_test) --> Check memory status after evict memory
04:48:30 DEBUG| (monitor qmp1) Sending command 'query-balloon'
04:48:30 DEBUG| Send command: {'execute': 'query-balloon', 'id': 'p7T7hEL2'}
04:48:30 DEBUG| (monitor qmp1) Response to 'query-balloon' (re-formated)
04:48:30 DEBUG| (monitor qmp1)    actual: 3835691008
04:48:30 DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s)
04:48:30 DEBUG| Sending command: wmic OS get FreePhysicalMemory
04:48:30 ERROR| Memory size mismatch after evict memory:

04:48:30 ERROR| Wanted to be changed: 438
Changed in monitor: 438
Changed in guest: 125

File: qemu/tests/balloon_service.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def memory_check(vm, get_polling_output, keyname):
 
         error_context.context("Compare memory from guest with qmp",
                               logging.info)
-        if (abs(guest_mem - stat_memory_qmp)) > (guest_mem * check_mem_ratio):
+        if abs(float(guest_mem - stat_memory_qmp)) / guest_mem > check_mem_ratio:
```
Comment (suqinhuang) [inline qemu/tests/balloon_service.py]: the methord is almost the same, so if issue is fixed because need to sleep for a while?

**Comment by CongLi** [comment]
Close this PR which has been fixed in c0e6db2255eaa83bd907232ae6d6ae09037d50c7

**Pull request closed without merging**